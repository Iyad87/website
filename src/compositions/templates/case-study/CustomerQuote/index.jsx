import React from 'react';
import PropTypes from 'prop-types';
import GatsbyImage from 'gatsby-image';
import {
  SubDescription,
  QuoteWrapper,
  Quote,
  Author,
  Position,
} from './styled';

const CustomerQuote = ({
  quotesImage,
  quotesImageAlt,
  quotes,
  quoteAuthor,
  quotePosition,
  subDescription,
  clientAppImage,
  clientAppImageAlt,
}) => (
  <div>
    <div>
      <GatsbyImage alt={quotesImageAlt} {...quotesImage.childImageSharp} />
      <QuoteWrapper>
        <Quote>{quotes}</Quote>
        <Author>{quoteAuthor}</Author>
        <Position>{quotePosition}</Position>
      </QuoteWrapper>
    </div>
    <SubDescription>{subDescription}</SubDescription>
    <GatsbyImage alt={clientAppImageAlt} {...clientAppImage.childImageSharp} />
  </div>
);

CustomerQuote.propTypes = {
  quotesImage: PropTypes.shape({
    childImageSharp: PropTypes.shape(GatsbyImage.propTypes).isRequired,
  }).isRequired,
  quotesImageAlt: PropTypes.string.isRequired,
  quotes: PropTypes.string.isRequired,
  quoteAuthor: PropTypes.string.isRequired,
  quotePosition: PropTypes.string.isRequired,
  subDescription: PropTypes.string.isRequired,
  clientAppImage: PropTypes.shape({
    childImageSharp: PropTypes.shape(GatsbyImage.propTypes).isRequired,
  }).isRequired,
  clientAppImageAlt: PropTypes.string.isRequired,
};

export default CustomerQuote;
