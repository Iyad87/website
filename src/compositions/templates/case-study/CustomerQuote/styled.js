import styled from 'styled-components';
import { highlight, textLighter, textOnDark } from 'styles/colours';

export const QuoteWrapper = styled.div`
  color: ${textOnDark};
  margin: -13% 7% 2% 7%;
  background-color: ${highlight};
  position: relative;
  font-weight: 300;
  line-height: 1.24;
`;

export const Quote = styled.p`
  font-size: 2.1rem;
  padding: 77px 111px 20px 107px;
`;

export const Author = styled.p`
  font-size: 1.1875rem;
  padding-left: 107px;
  margin-bottom: 0px;
  font-weight: 500;
`;

export const Position = styled.p`
  margin-top: 0px;
  font-size: 1.1875rem;
  padding-left: 107px;
  padding-bottom: 88px;
`;

export const SubDescription = styled.p`
  color: ${textLighter};
  font-size: 1.25rem;
  white-space: pre-line;
  margin-top: 109px;
  margin-bottom: 75px;
  font-weight: 300;
`;
