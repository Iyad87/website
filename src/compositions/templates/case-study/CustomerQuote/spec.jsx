import React from 'react';
import { shallow } from 'enzyme';

import CustomerQuote from './index';

test('CustomerQuote', () => {
  const props = {
    quotesImage: {
      childImageSharp: {
        src: 'image.jpg',
      },
    },
    quotesImageAlt: 'Quote image',
    quotes: 'YLD offered amazing service,',
    quoteAuthor: 'Casey Bisson',
    quotePosition: 'Senior Director of Product Management - Joyent',
    subDescription:
      'Some amazing support copy with go here to reenforce whatever section it is relevant too.',
    clientAppImage: {
      childImageSharp: {
        src: 'image.jpg',
      },
    },
    clientAppImageAlt: 'Client App Image',
  };
  const wrapper = shallow(<CustomerQuote {...props} />);

  expect(wrapper).toMatchSnapshot();
});
