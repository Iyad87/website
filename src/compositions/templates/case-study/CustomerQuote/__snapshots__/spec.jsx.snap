// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerQuote 1`] = `
<React.Fragment>
  <PageSection
    gradient={false}
    gradientHeight="50%"
    wide={true}
  >
    <Image
      Tag="div"
      alt="Quote image"
      fadeIn={true}
      src="image.jpg"
    />
    <Container>
      <styled__QuoteWrapper>
        <styled__Quote>
          YLD offered amazing service,
        </styled__Quote>
        <styled__Author>
          Casey Bisson
        </styled__Author>
        <styled__Position>
          Senior Director of Product Management - Joyent
        </styled__Position>
      </styled__QuoteWrapper>
    </Container>
  </PageSection>
  <PageSection
    gradient={false}
    gradientHeight="50%"
  >
    <Typography__base
      newLines={true}
    >
      Some amazing support copy with go here to reenforce whatever section it is relevant too.
    </Typography__base>
  </PageSection>
  <PageSection
    gradient={false}
    gradientHeight="50%"
    wide={true}
  >
    <Image
      Tag="div"
      alt="Client App Image"
      fadeIn={true}
      src="image.jpg"
    />
  </PageSection>
</React.Fragment>
`;
